superset-frontend/packages/superset-ui-chart-controls/src/operators/pivotOperator.ts

```diff
@@ -31,11 +31,13 @@ export const pivotOperator: PostProcessingFactory<PostProcessingPivot> = (
 ) => {
   const metricLabels = ensureIsArray(queryObject.metrics).map(getMetricLabel);
   const { x_axis: xAxis } = formData;
+
   if ((xAxis || queryObject.is_timeseries) && metricLabels.length) {
+    const index = [getColumnLabel(formData.x_axis || DTTM_ALIAS)];
     return {
       operation: 'pivot',
       options: {
-        index: [xAxis || DTTM_ALIAS],
+        index,
```

Comment:
I'm not sure which is better/more readable, inlining it or declaring it as a variable first. I went with declaring a `const` first, but looking at it now I should maybe have inlined it..

Reply:
I think this looks good 👍